pkl-codegen-java/src/main/kotlin/org/pkl/codegen/java/JavaCodeGenerator.kt

```diff
@@ -872,6 +869,15 @@ class JavaCodeGenerator(
       } else key
     }
   }
+
+  private fun mapPackageName(name: String): String =
+    codegenOptions.packageMapping[name] ?: name
+  
+  private fun mapModuleName(name: String): kotlin.Pair<String, String> {
+    val packageName = mapPackageName(name.substringBeforeLast('.', ""))
+    val className = name.substringAfterLast('.').replaceFirstChar { it.titlecaseChar() }
```

Comment:
This change would be different from the current behavior.
```
// current:
module foo.bar.baz -> package foo.bar; class Baz {}

// proposed:
module foo.bar.baz -> package foo.bar; class baz {}
```

That is, the class name would be lowercase, which is against Java naming conventions. This, in turn, would require users to always use a module name like `foo.bar.Baz` if they want to get conventional Java names.

Ultimately, I think that even if we decide to do this, this should happen as a separate change, because it is only tangentially related to the original purpose of this PR.

Reply:
Canonically, such Pkl code should also have `module foo.bar.Baz` if it's meant to be a class.

At the very least, though, if a package mapping is provided, we should probably not alter it.

Reply:
> Canonically, such Pkl code should also have module foo.bar.Baz if it's meant to be a class.

This is the first time I hear about it :( I don't believe it is documented anywhere, and it is definitely not followed by many Pkl modules in the wild.

Different behavior when mapping is specified or not would be very confusing, IMO. Also, the mapping is specifically about packages, not module names.

I would prefer to limit the scope of the change and avoid doing unexpected changes in the existing behavior. But if you insist I can change it here.

Reply:
It's part of our style guide actually; see [filename](https://pkl-lang.org/main/current/style-guide/index.html#filename) and [module name](https://pkl-lang.org/main/current/style-guide/index.html#module-name)

Reply:
Okay, this does make sense, thanks for the link! Still, I think that this change should not be done in this PR, since it is not related to its purpose.

Reply:
Ah, I see. Agree, makes sense to not touch it in this PR.